### PR TITLE
fix(condo): DOMA-10793 add validation for register service consumer

### PR DESCRIPTION
**Pull request by vovaaxeapolla**

### CODE REVIEW

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.js [hunk 1]
```diff
@@ -104,8 +104,13 @@ const RegisterServiceConsumerService = new GQLCustomSchema('RegisterServiceConsu
                 const { data: { dv, sender, resident: { id }, accountNumber, paymentCategory } } = args
 
                 if (!accountNumber || accountNumber.length === 0) throw new GQLError(ERRORS.ACCOUNT_NUMBER_IS_NOT_SPECIFIED, context)
-
-                const resident = await Resident.getOne(context, { id, deletedAt: null }, 'id user { id } address addressKey')
+                const isAdmin = get(context, 'authedItem.isAdmin')
```
Comment (dkoviazin) [inline apps/condo/domains/resident/schema/RegisterServiceConsumerService.js:107]: better to check if user type is Resident

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.js [hunk 2]
```diff
@@ -104,8 +104,13 @@ const RegisterServiceConsumerService = new GQLCustomSchema('RegisterServiceConsu
...
+                const residentWhere = {
```
Comment (pahaz) [inline apps/condo/domains/resident/schema/RegisterServiceConsumerService.js:108]: You need to check it in access

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.js [hunk 3]
```diff
@@ -223,7 +228,13 @@ const RegisterServiceConsumerService = new GQLCustomSchema('RegisterServiceConsu
                 if (!accountNumber || accountNumber.length === 0) {
                     throw new GQLError(ERRORS.ACCOUNT_NUMBER_IS_NOT_SPECIFIED, context)
                 }
-                const resident = await Resident.getOne(context, { id: residentId, deletedAt: null }, 'id user { id }')
+                const isAdmin = get(context, 'authedItem.isAdmin')
```
Comment (dkoviazin) [inline apps/condo/domains/resident/schema/RegisterServiceConsumerService.js:231]: isAdmin is a bad check as we can change access later
you need to add logic for Resident user

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.test.js [hunk 4]
```diff
@@ -132,6 +133,23 @@ describe('RegisterServiceConsumer', () => {
             expect(consumer1.id).not.toEqual(consumer2.id)
         })
```
Comment (dkoviazin) [inline apps/condo/domains/resident/schema/RegisterServiceConsumerService.test.js:135]: no positive check for admin

File: apps/condo/domains/resident/schema/RegisterServiceConsumerService.js (hunk 2 above)
Comment (pahaz) [reply apps/condo/domains/resident/schema/RegisterServiceConsumerService.js:108]: Or it’s better always use the same logic. It should not depend on isAdmin access

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/resident/access/RegisterServiceConsumerService.js [hunk 5]
```diff
@@ -1,17 +1,35 @@
...
+    if (!user) throwAuthenticationError()
     if (user.deletedAt) return false
+    if (user.isAdmin) return true
+    if (user.type === RESIDENT) {
+        //Two mutations have different input types
+        const residentId = get(data, 'residentId') || get(data, 'resident.id')
+
+        if (!residentId) return false
+
+        const [isMatch] = await find('Resident', {
```
Comment (dkoviazin) [inline apps/condo/domains/resident/access/RegisterServiceConsumerService.js:21]: you can use findOne if your condition has id

**Review by abshnko — APPROVED** [review]

**Pull request merged**